servers/src/main/java/tachyon/worker/file/FileWorkerMasterSyncExecutor.java

```diff
@@ -82,9 +82,9 @@ public void heartbeat() {
     }
     Preconditions.checkState(command.commandType == CommandType.Persist);
 
-    for (CheckpointFile checkpointFile : command.checkpointFiles) {
-      mFixedExecutionService.execute(new FilePersister(mLineageDataManager, checkpointFile.fileId,
-          checkpointFile.blockIds));
+    for (PersistFile persistFile : command.getCommandOptions().getPersistOptions().persistFiles) {
```

Comment:
Yeah, we could have precondition check, but its functionality is similar to throwing a NPE, right?